ios/algorithms.go

```diff
@@ -0,0 +1,55 @@
+package ios
+
+import (
+	"bytes"
+)
+
+// IndexWithTable returns the first index of `needle` found in `haystack`.
+// It needs the slide information of substr to accurately determine the index
+// using the Boyer-Moore algorithm. This algorithm is particularly useful for large needles.
+func IndexWithTable(haystack, substr []byte) int {
```

Comment:
in the func header it says `needle` but in the code it's `substr`, be consistent.